xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-io/src/main/java/org/xwiki/annotation/io/internal/DefaultIOService.java

```diff
@@ -240,11 +248,12 @@ public Annotation getAnnotation(String target, String annotationID) throws IOSer
             BaseObject object =
                 document.getXObject(this.configuration.getAnnotationClassReference(),
                     Integer.valueOf(annotationID));
-            if (object == null || !localTargetId.equals(object.getStringValue(Annotation.TARGET_FIELD))) {
+            String targetField = object == null ? null : object.getStringValue(Annotation.TARGET_FIELD);
+            if (object == null || !(Objects.equals(localTargetId, targetField) || Objects.equals("", targetField))) {
```

Comment:
Same here regarding `isBlank` and matching the empty string only for the document/its content and not all object properties.